packages/rum-core/src/domain/resource/resourceCollection.ts

```diff
@@ -41,13 +42,11 @@ export function startResourceCollection(
   lifeCycle: LifeCycle,
   configuration: RumConfiguration,
   pageStateHistory: PageStateHistory,
+  taskQueue = createTaskQueue(),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏What is the benefit of the taskQueue instead of directly using requestIdleCallback?

Reply:
Like, calling `requestIdleCallback` for each resource entry? This is a great question, 

* Performance: we have less asynchronous tasks to schedule, with task queues, tasks are run in synchronous batch.

* Scheduling: if each task use `requestIdleCallback` directly, each task will use its own timeout. So if we have a a lot of tasks timeout at the same time, we will still block the main thread for an uncontrolled amount of time. With the taskqueue, we can control how much time we block the main thread.

Ideally, we should have one taskqueue shared across the SDK, and replace`requestIdleRequest` usages with it, in particular in `mutationBatch` but it's not easy as there are other constraints here (= the throttle that you added in #2807), this is why I didn't act on it yet.

Reply:
Interesting! Sound good. I agree that ideally we should use this concept everywhere instead of requestIdleCallback. Maybe we can add it to our backlog.